### DISCUSSION TITLE
How do I create a plugin that brings a panoramic view with wayland fire?

### DISCUSSION THREAD

**Discussion by ghost** [WayfireWM/wayfire#1571] [Q&A] [answered]
Hi all. I would like to know how do I create a plugin that brings a panoramic view with wayland fire. So How do I create a plugin that brings a panoramic view with wayland fire?

For example... [Panolens/examples/panorama_infospot](https://pchen66.github.io/Panolens/examples/panorama_infospot.html) - source-code [pchen66/panolens.js](https://github.com/pchen66/panolens.js)

**Comment by soreau** [comment, marked as answer]
I'm not sure exactly what you have in mind, but wayfire cube plugin allows setting for skydome or cubemap images, so when you view the cube with ctrl+alt+drag, there is a background.

**Reply by ghost** [reply]
Hi soreau. 

> I'm not sure exactly what you have in mind, but wayfire cube plugin allows setting for skydome or cubemap images, so when you view the cube with ctrl+alt+drag, there is a background.

#### It would be something like this
<img src="https://miro.medium.com/max/900/0*NwoPwwcwjPpc5eqa.gif"/>

**Reply by soreau** [reply]
Sounds like maybe you want inside cube? Wayfire doesn't directly support this but you can adjust the Zoom setting of cube plugin (maybe -1.3) to bring the camera inside the cube.

**Reply by ghost** [reply]
> Sounds like maybe you want inside cube?

Yeah.

>  Wayfire doesn't directly support this but you can adjust the Zoom setting of cube plugin (maybe -1.3) to bring the camera inside the cube.

thank you for feedback.

**Reply by soreau** [reply]
If a patch existed to properly support inside cube, I don't see any reason why it couldn't be accepted upstream.
